Fix `[:, int]` indexing in `MultiEmbeddingTensor`

akihironitta opened the pull request:
Fixes a bug in `[:, int]` indexing in `MultiEmbeddingTensor` caught by this new test case:
```python
assert_equal(
        tensor_list,
        MultiEmbeddingTensor.cat(
            [met[:, i] for i in range(met.size(1))],
            dim=1,
        ),
    )
```

Addresses https://github.com/pyg-team/pytorch-frame/pull/193#discussion_r1387650926 (cc @weihua916)
Addresses https://github.com/pyg-team/pytorch-frame/pull/181#discussion_r1383949234 (cc @yiweny)

---

No changelog entry needed as the feature hasn't been released yet.

weihua916 commented on torch_frame/data/multi_embedding_tensor.py:
```diff
@@ -325,11 +325,18 @@ def _single_index_select(
         elif dim == 1:
             value_index = slice(self.offset[index], self.offset[index + 1])
+            values = self.values[:, value_index]
+            offset_list = [0, self.offset[index + 1] - self.offset[index]]
+            offset = torch.tensor(
+                offset_list,
+                dtype=torch.long,
+                device=self.device,
+            )
```
nit: offset = self.offset[[0, index+1]] - self.offset[[0, index]]

akihironitta commented:
@rusty1s When you have a chance, could you enable "Require conversation resolution before merging" for this repo to avoid accidentally auto-merging PRs before addressing review comments?

<img width="1266" alt="image" src="https://github.com/pyg-team/pytorch-frame/assets/20610905/f773ee05-6bce-445f-af0c-744981605ed1">

rusty1s commented:
Done.
